### PR TITLE
feat: Add ingress counter to avoid undesired deletion, Use networking…

### PR DESCRIPTION
**Pull request by Kirchen99**
feat: Add ingress counter to avoid undesired deletion, Use networking api instead of extentions for ingress, Add CNAME support, Recordset type configurable with parameter (by external expert on behalf of DB Systel GmbH)

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -8,20 +8,23 @@ import (
 	"sync"
 	"syscall"
 
-	"github.com/go-kit/kit/log/level"
-	"github.com/dbsystel/kube-controller-dbsystel-go-common/controller/ingress"
-	"github.com/dbsystel/kube-controller-dbsystel-go-common/kubernetes"
-	k8sflag "github.com/dbsystel/kube-controller-dbsystel-go-common/kubernetes/flag"
+	"github.com/Kirchen99/AmazonRoute53-ingress-controller/controller"
```
Comment (dbluxo) [inline cmd/main.go]: we have to use github.com/dbsystel....

File: controller/controller.go [hunk 2]
```diff
@@ -1,36 +1,44 @@
 package controller
 
 import (
-  "encoding/json"
+	"encoding/json"
 	"regexp"
 	"strconv"
 	"strings"
 
+	"github.com/Kirchen99/AmazonRoute53-ingress-controller/aws"
```
Comment (dbluxo) [inline controller/controller.go]: we have to use github.com/dbsystel....

File: go.mod [hunk 3]
```diff
@@ -1,37 +1,22 @@
-module github.com/dbsystel/AmazonRoute53-ingress-controller
```
Comment (dbluxo) [inline go.mod:1]: we have to use github.com/dbsystel....

**Review by dbluxo — CHANGES_REQUESTED** [review]

File: cmd/main.go (hunk 1 above)
Comment (Kirchen99) [reply cmd/main.go]: Yep

File: controller/controller.go (hunk 2 above)
Comment (Kirchen99) [reply controller/controller.go]: OK

File: go.mod (hunk 3 above)
Comment (Kirchen99) [reply go.mod:1]: good point

File: go.mod [hunk 4]
```diff
@@ -1,37 +1,21 @@
...
+module github.com/Kirchen99/AmazonRoute53-ingress-controller
```
Comment (dbluxo) [inline go.mod:1]: we have to use github.com/dbsystel....

**Review by dbluxo — CHANGES_REQUESTED** [review]

**Review by dbluxo — APPROVED** [review]

**Pull request merged**